src/src/PluginStructs/P123_data_struct.cpp

```diff
@@ -0,0 +1,707 @@
+#include "../PluginStructs/P123_data_struct.h"
+
+#ifdef USES_P123
+
+// the main constructor
+P123_data_struct::P123_data_struct() :
+  last_hum_val(0.0f),
+  last_temp_val(0.0f),
+  last_measurement(0),
+  state(P123_state::Uninitialized) {}
+
+//check if the plugin is initialized
+bool P123_data_struct::initialized() const {
+  return state != P123_state::Uninitialized;
+}
+
+
+//The main state machine
+//Only perform the measurements with big interval to prevent the sensor from warming up.
```

Comment:
Hmm this looks very familiar.
Is this copy/paste code?

Reply:
I think we worked together on another plugin so yes.